### PR TITLE
#62: added TTL Type handling

### LINKED ISSUE #62: introduce object for `Chain.ttl` param
can be either one of:
- `"FixedTTL(453800)"`
- `"RelativeTTL(50)"`

image

### PR DESCRIPTION
**Pull request by mitch-lbw**

### CODE REVIEW

File: src/test/java/com/kryptokrauts/codegen/maven/BaseTest.java [hunk 1]
```diff
@@ -48,16 +48,15 @@ public static void initConfig() throws MojoExecutionException {
...
-            .network(Network.LOCAL_IRIS_NETWORK)
+            .network(Network.TESTNET)
             .keyPair(keyPair)
             .targetVM(VirtualMachine.FATE)
             .compile();
   }
 
   protected static String getNodeBaseUrl() throws MojoExecutionException {
     String nodeBaseUrl = System.getenv(AETERNITY_BASE_URL);
+    nodeBaseUrl = "http://aelocal";
```
Comment (marc0olo) [inline src/test/java/com/kryptokrauts/codegen/maven/BaseTest.java]: ?

File: src/test/java/com/kryptokrauts/codegen/maven/BaseTest.java [hunk 2]
```diff
@@ -48,16 +48,15 @@ public static void initConfig() throws MojoExecutionException {
 
     aeternityServiceConfig =
         AeternityServiceConfiguration.configure()
-            .compilerBaseUrl(getCompilerBaseUrl())
-            .baseUrl(getNodeBaseUrl())
-            .network(Network.LOCAL_IRIS_NETWORK)
+            .network(Network.TESTNET)
```
Comment (marc0olo) [inline src/test/java/com/kryptokrauts/codegen/maven/BaseTest.java]: what's the reason for that change?

File: src/test/java/com/kryptokrauts/codegen/maven/BaseTest.java (hunk 1 above)
Comment (mitch-lbw) [reply src/test/java/com/kryptokrauts/codegen/maven/BaseTest.java]: Cleanup pending ;)

**Review by marc0olo — APPROVED** [review]
need to review this in use with contract-maven-showcase. but generally looks good to me

**Pull request merged**